feature/experimental/p4rt/tests/p4rt_election/p4rt_election_test.go

```diff
@@ -0,0 +1,636 @@
+	desc            string
+	lowID           uint64
+	highID          uint64
+	deviceID        uint64
+	handle          *p4rt_client.P4RTClient
+	wantStatus      int32
+	wantFinalStatus int32
+	wantFail        bool
+	wantWrite       bool
+	wantRead        bool
+}
+
+// configureDeviceId configures p4rt device-id on the DUT.
+func configureDeviceId(t *testing.T, dut *ondatra.DUTDevice) {
+	nodes := p4rtutils.P4RTNodesByPort(t, dut)
+	p4rtNode, ok := nodes["port1"]
+	if !ok {
+		t.Fatal("Couldn't find P4RT Node for port: port1")
+	}
+	t.Logf("Configuring P4RT Node: %s", p4rtNode)
+	c := oc.Component{}
+	c.Name = ygot.String(p4rtNode)
+	c.IntegratedCircuit = &oc.Component_IntegratedCircuit{}
+	c.IntegratedCircuit.NodeId = ygot.Uint64(deviceId)
+	gnmi.Replace(t, dut, gnmi.OC().Component(p4rtNode).Config(), &c)
+}
+
+// configurePortId configures p4rt port-id on the DUT.
+func configurePortId(t *testing.T, dut *ondatra.DUTDevice) {
+	portName := dut.Port(t, "port1").Name()
+	gnmi.Replace(t, dut, gnmi.OC().Interface(portName).Id().Config(), portId)
+}
+
+// Create client connection
+func clientConnection(t *testing.T, dut *ondatra.DUTDevice) *p4rt_client.P4RTClient {
+	clientHandle := p4rt_client.NewP4RTClient(&p4rt_client.P4RTClientParameters{})
+	if err := clientHandle.P4rtClientSet(dut.RawAPIs().P4RT().Default(t)); err != nil {
+		t.Fatalf("Could not initialize p4rt client: %v", err)
+	}
+	return clientHandle
+}
+
+// Setup P4RT Arbitration Stream which will return a GRPC status code
+// in line with the client arbitration notifications. An error can also
+// be returned.
+func streamP4RTArb(args *testArgs) (int32, error) {
+	streamParameter := p4rt_client.P4RTStreamParameters{
+		Name:        streamName,
+		DeviceId:    args.deviceID,
+		ElectionIdH: args.highID,
+		ElectionIdL: args.lowID,
+	}
+	// Send ClientArbitration message for a given handle.
+	if args.handle != nil {
+		args.handle.StreamChannelCreate(&streamParameter)
+		if err := args.handle.StreamChannelSendMsg(&streamName, &p4_v1.StreamMessageRequest{
+			Update: &p4_v1.StreamMessageRequest_Arbitration{
+				Arbitration: &p4_v1.MasterArbitrationUpdate{
+					DeviceId: streamParameter.DeviceId,
+					ElectionId: &p4_v1.Uint128{
+						High: streamParameter.ElectionIdH,
+						Low:  streamParameter.ElectionIdL,
+					},
+				},
+			},
+		}); err != nil {
+			return 0, fmt.Errorf("errors seen when sending ClientArbitration message: %v", err)
+		}
+		time.Sleep(1 * time.Second)
+		return getRespCode(args)
+	}
+	return 0, errors.New("Missing Client")
+}
+
+// Obtain status Code from Arbitration
+func getRespCode(args *testArgs) (int32, error) {
+	handle := args.handle
+	if handle != nil {
+		// Grab Arb Response to look at status code
+		_, arbResp, arbErr := handle.StreamChannelGetArbitrationResp(&streamName, 1)
+		if arbErr != nil {
+			if err := p4rtutils.StreamTermErr(args.handle.StreamTermErr); err != nil {
+				if strings.Contains(err.Error(), "InvalidArgument") {
+					return 3, err
```

Comment:
Please use status and code package here. 

https://pkg.go.dev/google.golang.org/grpc/status

Reply:
sounds good, will add

Reply:
Just to make sure I'm not going off the deep-end and adhering to the standard, you mean something like this, correct?

```go
				if status.Code(err) == codes.InvalidArgument {
					return int32(codes.InvalidArgument), err
				}
```

Reply:
No need to validate `InvalidArgument` here. Just return whatever error was returned by the server. 

```
If err != nil {
   return int32(status.Code(err))
}
```

Reply:
SG - Thanks for clarification